src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -1272,6 +1272,15 @@ GraphPatternOperation Visitor::visitSpatialQuery(
   return spatialQuery;
 }
 
+// _____________________________________________________________________________
+GraphPatternOperation Visitor::visitMaterializedViewQuery(
+    Parser::ServiceGraphPatternContext* ctx) {
+  parsedQuery::MaterializedViewQuery viewQuery;
+  parseBodyOfMagicServiceQuery(viewQuery, ctx, "materialized view query");
+  return viewQuery;
```

Comment:
We have no very many of these functions that look the same. 
For some of them the only difference is that they have a `try {query.validate()} catch{reportError}`.

Can you create a unified templated member function for this that is just templated on the concrete subclass of the magicServiceQuery (therefore you would probably need the `validate()` method in the base class and adapt some of the other magic queries, but as those are coming out left and right, this would be worth it.